src/pages/Users/Users.jsx

```diff
@@ -23,19 +27,27 @@ function Users() {
         dispatch( setUsers(usersData) )
       }
     }
-    if (token.length) {
+    if (token && token.length) {
       getUsers()
     }
   }, [])
   useEffect(() => {
-    dispatch( setFilteredAndSortedUsers([...users]) )
-  }, [users])
+    let filteredUsers = helper.filterUsers(filter, users)
+    setFilteredAndSortedUsers( helper.sortUsers(selectedSort, filteredUsers) )
+  }, [users, filter, selectedSort])
```

Comment:
As I mentioned in issue #1, `filteredAndSortedUsers` should not be in state. Also, this effect is not needed.

As recommended by the official React documentation, it's better to calculate `filteredAndSortedUsers` during rendering. Because even though you fixed the bug with lost filtering and sorting, keeping `filteredAndSortedUsers` in state can lead to new bugs in the future.

See https://react.dev/learn/you-might-not-need-an-effect#updating-state-based-on-props-or-state for more details (I reocmmend reading the whole article).